### PR TITLE
refactor(payments): update CouponForm to utilize Tailwind

### PR DESCRIPTION
**Pull request by sardesam**
## Because

- We want to convert to Tailwind across fxa for consistency and to improve overall maintenance
- We want to improve responsiveness and UI/UX
- We want to improve accessibility

## This pull request

- Replaces all CSS/SCSS with Tailwind utility classes with the exception of:
  - shared styles associated with form input, CTAs, and colours
- Removes styles from the .scss file specific to the `CouponForm` component, not including what is mentioned above
- Reworks Storybook files

## Issue that this pull request solves

Closes: #13796/FXA-5625 and FXA-5845

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx [hunk 1]
```diff
@@ -187,23 +187,31 @@ export const CouponForm = ({
...
-              className={`button ${readOnly ? 'hidden' : ''}`}
-              onClick={removeCoupon}
-              disabled={subscriptionInProgress}
-              data-testid="coupon-remove-button"
-            >
-              <Localized id="coupon-remove">
-                <span>Remove</span>
-              </Localized>
-            </button>
+            <div className="w-28 ml-4">
```
Comment (pdehaan) [inline packages/fxa-payments-server/src/components/CouponForm/index.tsx:196]: Per https://tailwindcss.com/docs/width#fixed-widths, setting `w-28`, sets `width: 7rem; /* 112px */`.
Is that risky if this is localized and some languages (ie: German) can be fairly "wordy"?

For giggles, I ran this in the fxa-content-server-l10n repo: <kbd>git grep -h "coupon-remove" | sort | uniq</kbd>
Then if I pipe it through some script which sorts by line length, we see that "Remove" is the 28th longest string (by character count, but not all character widths are equal, so it's a bit of a lame metric versus measured text width):

```sh
 1. coupon-remove = Eltávolítás
 2. coupon-remove = Verwijderen
 3. coupon-remove = Allontanar
 4. coupon-remove = Fuortsmite
 5. coupon-remove = Wotstronić
 6. coupon-remove = Wótwónoźeś
 7. coupon-remove = Премахване
 8. coupon-remove = Entfernen
 9. coupon-remove = Fjarlægja
10. coupon-remove = Odstrániť
…
28. coupon-remove = Remove
…
39. coupon-remove = 삭제
```

Where <kbd>git grep "Eltávolítás"</kbd> tells me that the string is Hungarian:
```
locale/hu/main.ftl:coupon-remove = Eltávolítás
```

If I hack that string into staging payments via DevTools it _just_ seems to fit, but adding 2-3 "m"s at the end causes it to start clipping text at both ends. But maybe it's a non-issue if we were hardcoding it to 110px earlier. 🤷 

Screen Shot 2022-08-12 at 9 38 31 AM

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx [hunk 2]
```diff
@@ -187,23 +187,31 @@ export const CouponForm = ({
...
+              <button
+                className={`button ${readOnly ? 'hidden' : ''}`}
```
Comment (pdehaan) [inline packages/fxa-payments-server/src/components/CouponForm/index.tsx:198]: Not sure what/if `readOnly` is checking here. I think we already check/abort a `readOnly` check above on L195 where if we _are_ `readOnly` we don't render this block, no?

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx [hunk 3]
```diff
@@ -222,22 +230,25 @@ export const CouponForm = ({
...
-            className="button"
-            type="submit"
-            data-testid="coupon-button"
-            disabled={checkingCoupon || readOnly || subscriptionInProgress}
-          >
-            <Localized id="coupon-submit">
-              <span>Apply</span>
-            </Localized>
-          </button>
+          <div className="w-28 ml-4">
```
Comment (pdehaan) [inline packages/fxa-payments-server/src/components/CouponForm/index.tsx:233]: Maybe same here. Hardcoding container to 112px wide with the following potential translations/strings:
```sh
 1. coupon-submit = Застосувати
 2. coupon-submit = వర్తింపజేయి
 3. coupon-submit = Appliquer
 4. coupon-submit = Applitgar
 5. coupon-submit = Toepassen
 6. coupon-submit = Прилагане
 7. coupon-submit = Применить
 8. coupon-submit = Alkalmaz
 9. coupon-submit = Anwenden
10. coupon-submit = Applicar
…
32. coupon-submit = Apply
…
41. coupon-submit = 적용
```

```js
// From within the cloned "fxa-content-server-l10n` repo folder.
const cp = require("child_process");

const QUERY = "coupon-submit =";
const res = cp.execSync(`git grep -h "${QUERY}" | sort | uniq`)
  .toString()
  .trim();

const lines = res.split("\n")
  .sort((a, b) => b.length - a.length);
console.log(lines.map((val, idx) => `${String(idx + 1).padStart(2, " ")}. ${val}`).join("\n"));
```

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx (hunk 2 above)
Comment (sardesam) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: Nice catch! Thank you! Fixed.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx (hunk 1 above)
Comment (sardesam) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: Excellent point! I removed the width restriction. I temporarily added extra padding to the sides as `button` and link elements will be addressed in FXA-5637.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx (hunk 3 above)
Comment (sardesam) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: Same here (more details in other comment). Thank you for pointing this out!

File: packages/fxa-payments-server/public/locales/en-US/main.ftl [hunk 4]
```diff
@@ -381,6 +381,8 @@ plan-details-list-price = List Price
 
 coupon-discount = Discount
 coupon-discount-applied = Discount Reward Applied
+coupon-promo-code = Promo Code
+coupon-promo-code-applied = Promo Code Applied
```
Comment (bcolsson) [inline packages/fxa-payments-server/public/locales/en-US/main.ftl:385]:
```suggestion
# Promo code for a coupon that provides a discount when applied
coupon-promo-code = Promo Code
# Promo code for a coupon that provides a discount when applied
coupon-promo-code-applied = Promo Code Applied
```
Suggest adding some individual comments as "Promo" could result in incorrect translations without context.

**Review by bcolsson — APPROVED** [review]

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx [hunk 5]
```diff
@@ -170,44 +170,48 @@ export const CouponForm = ({
...
       ) : (
-        <div className="coupon-header">
-          <h4>
-            <Localized id="coupon-discount">Discount</Localized>
-          </h4>
-        </div>
+        <h4 className="m-0 mb-4">
+          <Localized id="coupon-promo-code">Promo Code</Localized>
+        </h4>
       )}
```
[the remark this answers is not in the document]
Comment (sardesam) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: Great catch. Thank you! Updated.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx [hunk 6]
```diff
@@ -226,22 +228,25 @@ export const CouponForm = ({
...
+              <Localized id="coupon-submit">Apply</Localized>
+            </button>
+          </div>
         </form>
       )}
       {error && (
         <Localized id={error}>
-          <div className="coupon-error" data-testid="coupon-error">
+          <div
+            className="text-red-700 mt-4 w-[calc(100%-8rem)]"
```
Comment (xlisachan) [inline packages/fxa-payments-server/src/components/CouponForm/index.tsx:247]: `w-[calc(100%-8rem)]`?
With `w-[calc(100%-8rem)]`:
Screen Shot 2022-10-20 at 4 13 19 PM

Without `w-[calc(100%-8rem)]`:
Screen Shot 2022-10-20 at 4 14 07 PM

**Review by xlisachan — APPROVED** [review]
LGTM! 🐬

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx [hunk 7]
```diff
@@ -170,44 +170,48 @@ export const CouponForm = ({
 
   return (
     <div
-      className={`coupon-component ${coupon ? 'has-coupon' : ''}`}
+      className={`p-4 text-base mt-6 w-full bg-white rounded-lg shadow-sm shadow-grey-300 mobileLandscape:m-0 mobileLandscape:mt-6 tablet:my-8 tablet:max-w-xs coupon-component ${
+        coupon ? 'has-coupon' : ''
```
[the remark this answers is not in the document]
Comment (sardesam) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: Legacy code. Good catch. Removed.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx (hunk 6 above)
Comment (sardesam) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: The previous styles on the error included a restricted width to approximately the length of the textfield, but agreed that it looks better taking up the full width. Removed!

**Pull request merged**